code/HabitTracker/app/src/main/java/com/example/habittracker/activities/tracking/ProgressUpdater.java

```diff
@@ -0,0 +1,49 @@
+package com.example.habittracker.activities.tracking;
+
+import android.util.Log;
+
+import com.example.habittracker.DatabaseManager;
+import com.example.habittracker.Habit;
+import com.example.habittracker.HabitEvent;
+import com.example.habittracker.utils.HabitEventListCallback;
+import com.example.habittracker.utils.SharedInfo;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+
+public class ProgressUpdater {
```

Comment:
I know, I'm sorry for being the javadoc police. Please javadoc on your constructor and on the class

Reply:
Done